pkg/diagnose/check.go

```diff
@@ -93,30 +177,9 @@ func diagnose(diagCfg diagnosis.Config) []diagnosis.Diagnosis {
 		checkName := diagnoseConfig.Name
 		instances := collector.GetChecksByNameForConfigs(checkName, diagnoseConfigs)
 		for _, instance := range instances {
-
-			instanceName := string(instance.ID())
-
-			// Run check
-			runErr := instance.Run()
-
-			// Get diagnoses
-			checkDiagnoses, diagnoseErr := instance.GetDiagnoses()
-			if diagnoseErr == nil && len(checkDiagnoses) > 0 {
-				for _, d := range checkDiagnoses {
-					if len(d.Category) == 0 {
-						d.Category = instanceName
-					}
-					diagnoses = append(diagnoses, d)
-				}
-			} else if diagnoseErr != nil {
-				// Check Diagnose method return error
-				diagnoses = append(diagnoses, diagnosis.Diagnosis{
-					Result:    diagnosis.DiagnosisUnexpectedError,
-					Name:      instanceName,
-					Category:  instanceName,
-					Diagnosis: "Check Dianose failes with unexpected errors",
-					RawError:  runErr,
-				})
+			instanceDiagnoses := getInstanceDiagnoses(instance)
+			if len(instanceDiagnoses) > 0 {
+				diagnoses = append(diagnoses, instanceDiagnoses...)
 			}
```

Comment:
The condition is redundant, appending an empty slice is well-defined to do nothing:
```suggestion
			diagnoses = append(diagnoses, instanceDiagnoses...)
```

Reply:
Did not realized that, fixed, thank you